qutebrowser/keyinput/basekeyparser.py

```diff
@@ -366,6 +365,7 @@ def on_keyconfig_changed(self, mode):
 
     def clear_keystring(self):
         """Clear the currently entered key sequence."""
-        self._debug_log("discarding keystring '{}'.".format(self._keystring))
-        self._keystring = ''
-        self.keystring_updated.emit(self._keystring)
+        if self._keystring:
+            self._debug_log("discarding keystring '{}'.".format(self._keystring))
+            self._keystring = ''
+            self.keystring_updated.emit(self._keystring)
```

Comment:
I'm not 100% sure... FWIW, here's what happens when it gets emitted:

- `HintManager.handle_partial_key` filters hints
- For normal mode, the keystring in the statusbar gets updated
- `KeyHintView.update_keyhint` updates the key hint widget